test/functional/rpc_setban.py

```diff
@@ -36,27 +36,40 @@ def run_test(self):
         self.restart_node(1, ['-whitelist=127.0.0.1'])
         self.connect_nodes(0, 1)
         peerinfo = self.nodes[1].getpeerinfo()[0]
-        assert('noban' in peerinfo['permissions'])
+        assert "noban" in peerinfo["permissions"]
 
         # If we remove the ban, Node 0 should be able to reconnect even without noban permission
         self.nodes[1].setban("127.0.0.1", "remove")
         self.restart_node(1, [])
         self.connect_nodes(0, 1)
         peerinfo = self.nodes[1].getpeerinfo()[0]
-        assert(not 'noban' in peerinfo['permissions'])
+        assert not "noban" in peerinfo["permissions"]
 
         self.log.info("Test that a non-IP address can be banned/unbanned")
         node = self.nodes[1]
         tor_addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion"
         ip_addr = "1.2.3.4"
-        assert(not self.is_banned(node, tor_addr))
-        assert(not self.is_banned(node, ip_addr))
+        assert not self.is_banned(node, tor_addr)
+        assert not self.is_banned(node, ip_addr)
+
         node.setban(tor_addr, "add")
-        assert(self.is_banned(node, tor_addr))
-        assert(not self.is_banned(node, ip_addr))
+        assert self.is_banned(node, tor_addr)
+        assert not self.is_banned(node, ip_addr)
+
+        # Test that the ban list is preserved through restart.
+
+        self.restart_node(1)
+        assert self.is_banned(node, tor_addr)
+        assert not self.is_banned(node, ip_addr)
+
         node.setban(tor_addr, "remove")
-        assert(not self.is_banned(self.nodes[1], tor_addr))
-        assert(not self.is_banned(node, ip_addr))
+        assert not self.is_banned(self.nodes[1], tor_addr)
+        assert not self.is_banned(node, ip_addr)
+
+        self.restart_node(1)
+        assert not self.is_banned(self.nodes[1], tor_addr)
```

Comment:
As we have https://github.com/bitcoin/bitcoin/blob/88b843e49c84e331e40bfbe249269deaf2a6a2ac/test/functional/rpc_setban.py#L49

what are reasons to use `self.nodes[1]`? Did I miss something?

Reply:
Or, I see https://github.com/bitcoin/bitcoin/pull/20966/files#r637932195